### PR TITLE
Add cross-references between [basic.life],

### PR DESCRIPTION
**Pull request by ecatmur**
[class.cdtor] and [except.handle].

To resolve confusion noted in
https://github.com/cplusplus/CWG/issues/204

### CODE REVIEW

**Comment by AlisdairM** [comment]
Now that pseudo-destructors also end the lifetime of scalar objects, is there a latent Core issue lurking here?

**Comment by ecatmur** [comment]
> Now that pseudo-destructors also end the lifetime of scalar objects, is there a latent Core issue lurking here?

I don't feel there is. A scalar pseudo-destructor is necessarily trivial, so [class.cdtor] isn't needed, and [basic.life] is written in terms of destruction of the object, which is what a pseudo-destructor does https://eel.is/c++draft/expr.call#5.sentence-2 . Am I missing something?

**Comment by tkoeppe** [comment]
These references may be helpful. @jensmaurer do you think there could be a more concise way of placing those references?

File: source/basic.tex [hunk 1]
```diff
@@ -3451,6 +3451,9 @@
   the pointer is used as the operand of a
   \keyword{dynamic_cast}\iref{expr.dynamic.cast}.
 \end{itemize}
+\begin{note}
+For an object with a non-trivial constructor or destructor, see also~\ref{class.cdtor}.
+\end{note}
```
Comment (jensmaurer) [inline source/basic.tex:3456]: We already have this cross-reference in line 3425:

> For an object under construction or destruction, see~\ref{class.cdtor}.

In general, [class.cdtor] describes behavior for objects under construction or destruction, and [basic.life] describes behavior for objects that are (entirely) out-of-lifetime.

File: source/basic.tex [hunk 2]
```diff
@@ -3501,6 +3504,9 @@
 \keyword{dynamic_cast}\iref{expr.dynamic.cast} or as the operand of
 \keyword{typeid}.
 \end{itemize}
+\begin{note}
+For an object with a non-trivial constructor or destructor, see also~\ref{class.cdtor}.
+\end{note}
```
Comment (jensmaurer) [inline source/basic.tex:3509]: Same here, see line 3492.

File: source/basic.tex (hunk 1 above)
Comment (jensmaurer) [reply source/basic.tex:3456]: And yes, "object" might be a subobject.

**Comment by jensmaurer** [comment]
@tkoeppe, I'm not happy about the verbose notes, and I'm not sure they clarify anything.

If there is some problem with the wording (e.g. because [class.cdtor] wants to be in [basic.life]), we should entertain that as a CWG issue with a specific suggestion for a change.

**Comment by tkoeppe** [comment]
@ecatmur Could you please file a CWG issue if there is a need for clarification?

**Pull request closed without merging**